kuka_rsi_hw_interface/src/robot_control_node.cpp

```diff
@@ -22,76 +22,134 @@
 #include "kuka_rsi_hw_interface/robot_control_node.hpp"
 #include "kroshu_ros2_core/ROS2BaseLCNode.hpp"
 
-
 namespace kuka_rsi_hw_interface
 {
+  RobotControlNode::RobotControlNode(
+      const std::string &node_name,
+      const rclcpp::NodeOptions &options)
+      : kroshu_ros2_core::ROS2BaseLCNode(node_name, options)
+  {
+    auto qos = rclcpp::QoS(rclcpp::KeepLast(1));
+    auto callback =
+        [this](sensor_msgs::msg::JointState::SharedPtr msg)
+    { this->commandReceivedCallback(msg); };
+
+    joint_command_subscription_ = this->create_subscription<sensor_msgs::msg::JointState>(
+        "rsi_joint_command", qos, callback);
+
+    joint_state_publisher_ = this->create_publisher<sensor_msgs::msg::JointState>("rsi_joint_state", 1);
+    joint_command_msg_ = std::make_shared<sensor_msgs::msg::JointState>();
+  }
 
-RobotControlNode::RobotControlNode(
-  const std::string & node_name,
-  const rclcpp::NodeOptions & options)
-: kroshu_ros2_core::ROS2BaseLCNode(node_name, options)
-{}
-
-rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn RobotControlNode::
-on_configure(
-  const rclcpp_lifecycle::State &)
-{
-  kuka_rsi_hw_interface_ = std::make_unique<KukaHardwareInterface>(this->shared_from_this());
-  kuka_rsi_hw_interface_->configure();
-  return SUCCESS;
-}
+  CallbackReturn RobotControlNode::on_configure(const rclcpp_lifecycle::State &)
+  {
+    registerParameter<std::string>("rsi_ip_address_", "127.0.0.1",
```

Comment:
ip and port should not be allowed to be changed in inactivate state (2. value in ParameterSetAccessRights should be false)

Reply:
also true for n_nof